pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -116,6 +119,11 @@ public String toExplainString() {
   protected TransferableBlock getNextBlock() {
     try {
       consumeInputBlocks();
+
+      if (!_readyToConstruct) {
+        return TransferableBlockUtils.getNoOpTransferableBlock();
+      }
+
       return produceAggregatedBlock();
```

Comment:
wondering if we can wrapped around these with a BaseTransferrableOperator than handles metadata block (NOOP, ERROR, not sure if possible for EOS)

Reply:
se comment below, it isn't straightforward to make this clean (there isn't even a clean way to get the child block because you might have to know whether to read from left/right in case of a join or whether to read at all in the case that the operator is currently a 0->1 operator)